integration/experiment/ffnet/neural_net_sweep.py

```diff
@@ -0,0 +1,177 @@
+                new_freq_settings.append(new_val)
+        freq_settings = new_freq_settings
+    return freq_settings
+
+def report_signals(domains):
+    signals = []
+    if 'cpu' in domains:
+        signals += ["CPU_CYCLES_THREAD@package",
+                        "CPU_CYCLES_REFERENCE@package",
+                        "TIME@package",
+                        "CPU_ENERGY@package"]
+    if 'gpu' in domains:
+        signals += ["GPU_CORE_FREQUENCY_STATUS@board",
+                        "GPU_CORE_FREQUENCY_MIN_CONTROL@board",
+                        "GPU_CORE_FREQUENCY_MAX_CONTROL@board"]
+    return signals
+
+def trace_signals(domains):
+    signals = []
+    if 'cpu' in domains:
+        signals += ["CPU_POWER@package",
+                        "DRAM_POWER@package",
+                        "CPU_FREQUENCY_STATUS@package",
+                        "CPU_PACKAGE_TEMPERATURE@package",
+                        "MSR::UNCORE_PERF_STATUS:FREQ@package",
+                        "MSR::QM_CTR_SCALED_RATE@package",
+                        "CPU_INSTRUCTIONS_RETIRED@package",
+                        "CPU_CYCLES_THREAD@package",
+                        "CPU_ENERGY@package",
+                        "MSR::APERF:ACNT@package",
+                        "MSR::MPERF:MCNT@package",
+                        "MSR::PPERF:PCNT@package"]
+
+    if 'gpu' in domains:
+        signal += ["GPU_CORE_FREQUENCY_STATUS@gpu",
+                        "GPU_POWER@gpu",
+                        "GPU_UTILIZATION@gpu",
+                        "GPU_CORE_ACTIVITY@gpu",
+                        "GPU_UNCORE_ACTIVITY@gpu"]
+    return signals
+
+def launch_configs(output_dir, app_conf, freq_range):
+    agent = 'frequency_map'
+    targets = []
+    policy_names = {'cpu':'FREQ_CPU_DEFAULT',
+               'cpu_uncore':'FREQ_CPU_UNCORE',
+               'gpu':'FREQ_GPU_DEFAULT'}
+
+    freq_settings = explode_freq_settings(freq_range)
+    print(freq_settings)
+
+    for setting in freq_settings:
+        print(f"Setting: {setting}")
+        name = []
+        options = {}
+        for domain, freq in setting.items():
+            name.append(f"{domain}{freq:.1e}")
+            options[policy_names[domain]] = freq
+        name_str = "_".join(name)
+        file_name = os.path.join(output_dir, f'{agent}_agent_{name_str}.config')
+        agent_conf = geopmpy.agent.AgentConf(file_name, agent, options)
+        targets.append(launch_util.LaunchConfig(app_conf=app_conf,
+                                                agent_conf=agent_conf,
+                                                name=name_str))
+    return targets
+
+def launch(app_conf, args, experiment_cli_args):
+    '''
+    Run the application over a range of core, uncore, and gpu frequencies.
+    '''
+
+    output_dir = os.path.abspath(args.output_dir)
+    mach = machine.init_output_dir(output_dir)
+    freq_range = {}
+
+    if hasattr(args, 'min_frequency') and hasattr(args, 'max_frequency'):
```

Comment:
Minor nit just so the code reads more fluidly:
```suggestion
    if 'min_frequency' in args and 'max_frequency' in args:
```

Same comment applies below.